flow/FastAlloc.cpp

```diff
@@ -224,25 +224,35 @@ struct FastAllocator<Size>::GlobalData {
 	long long totalMemory;
 	long long partialMagazineUnallocatedMemory;
 	long long activeThreads;
-	GlobalData() : totalMemory(0), partialMagazineUnallocatedMemory(0), activeThreads(0) { 
+	GlobalData() : totalMemory(0), partialMagazineUnallocatedMemory(0), activeThreads(0) {
 		InitializeCriticalSection(&mutex);
 	}
 };
 
 template <int Size>
 long long FastAllocator<Size>::getTotalMemory() {
-	return globalData()->totalMemory;
+	EnterCriticalSection(&globalData()->mutex);
+	long long total = globalData()->totalMemory;
+	LeaveCriticalSection(&globalData()->mutex);
+	return total;
 }
 
 // This does not include memory held by various threads that's available for allocation
 template <int Size>
 long long FastAllocator<Size>::getApproximateMemoryUnused() {
-	return globalData()->magazines.size() * magazine_size * Size + globalData()->partialMagazineUnallocatedMemory;
+	EnterCriticalSection(&globalData()->mutex);
+	long long unused =
+	    globalData()->magazines.size() * magazine_size * Size + globalData()->partialMagazineUnallocatedMemory;
```

Comment:
This one might make sense (is size() threadsafe?), or maybe we care about the race where magazines.size() changes, but partialMagazineUnallocatedMemory hasn't been updated yet.